### PR TITLE
Closes #195 Add BRTHDTC to dm & dm_metabolic

### LINKED ISSUE #195: Documentation:  Add BRTHDTC to pharmaversesdtm::dm
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

Add `BRTHDTC` to `dm.R`. Check if this change is affecting to `dm_peds.R`, `dm_vaccine.R`, `dm_metabolic.R` files as well.

### PR DESCRIPTION
**Pull request by Lina2689**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
@manciniedoardo can you take over reviewing this please

File: data-raw/dm.R [hunk 1]
```diff
@@ -12,8 +15,82 @@ raw_suppdm <- read_xpt(paste0(sdtm_path, "suppdm", ".xpt?raw=true"))
...
+    dm <- dm %>% mutate(BRTHDTC = coalesce(BRTHDTC, as.character(NA)))
+    if (any(is.na(dm$BRTHDTC))) {
+      dm$BRTHDTC[is.na(dm$BRTHDTC)] <- fallback_dummy_brthdtc
+    }
+  }
+} else {
+  # Ensure character type for consistency
+  dm <- dm %>% mutate(BRTHDTC = as.character(BRTHDTC))
+}
+
```
Comment (Fanny-Gautier) [inline data-raw/dm.R]: in final `DM` data, could `BRTHDTC` be placed before  `AGE` variable, as in IG?

**Review by Fanny-Gautier — APPROVED** [review]
Fine for me, it can be merged if the minor comments do not need to be implemented.

File: data-raw/dm.R [hunk 2]
```diff
@@ -4,6 +4,9 @@
 # Load libraries -----
 library(haven)
 library(admiral)
+library(dplyr) # <- make sure this is loaded for mutate, coalesce, left_join
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply data-raw/dm.R]: will remove

File: data-raw/dm.R [hunk 3]
```diff
@@ -12,8 +15,85 @@ raw_suppdm <- read_xpt(paste0(sdtm_path, "suppdm", ".xpt?raw=true"))
...
+  na_idx <- is.na(out) & !is.na(x)
+  if (any(na_idx)) out[na_idx] <- suppressWarnings(parse_date_time(x[na_idx], orders = c("Y-m-dTH:M:S", "Y-m-d H:M:S", "Ymd"), tz = "UTC"))
+  out
+}
+
+# Add BRTHDTC to dm when missing ----
+# 1. Prefer existing BRTHDTC if present in raw_dm or dm.
+# 2. If missing, attempt to compute approximate birth date = anchor_date - AGE (when AGEU indicates years).
+#    anchor_date chosen from DMDTC, RFSTDTC, RFICDTC, RFXSTDTC (in that order).
+# 3. If still missing, set a fallback dummy value (configurable below).
```
Comment (bms63) [inline data-raw/dm.R]: Does this happen with our data?

**Comment by bms63** [comment]
Hi @Lina2689 just checking in on this one?  Will you have time to finish it out.  I think the extra checks you added are not needed just the simple stuff :)  It is like 98% there!!

**Review by bms63 — APPROVED** [review]
Awesome!!  Thanks @Lina2689 you rock!!

Let's merge this in and start to see what happens :)
